core/frontend/src/components/vehiclesetup/overview/OnboardSensors.vue

```diff
@@ -96,6 +98,40 @@ export default Vue.extend({
         .filter((param) => param.value !== 0)
         .map((parameter) => decode(parameter.name, parameter.value))
     },
+    compass_description(): Dictionary<string> {
+      const results = {} as Dictionary<string>
+      for (const compass of this.compasses) {
+        // First we check the priority for this device
+        let priority = 'Unused'
+        let number_in_parameter = 0
+        for (const param of autopilot_data.parameterRegex('^COMPASS_PRIO.*_ID')) {
+          if (param.value === compass.paramValue) {
+            const number_in_parameter_as_string = param.name.match(/\d+/g)?.[0] ?? '1'
+            number_in_parameter = parseInt(number_in_parameter_as_string, 10)
+            switch (number_in_parameter) {
+              case 1:
+                priority = '1st'
+                break
+              case 2:
+                priority = '2nd'
+                break
+              case 3:
+                priority = '3rd'
+                break
+              default:
+                priority = 'Unused'
```

Comment:
Should we use [ordinal](https://www.npmjs.com/package/ordinal) ? Os it'll not be greater than 3 ?

Reply:
Ardupilot doesn't support more than 3 compasses in use